### PR TITLE
feat(ctb): `OptimismPortal2` test migration

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Migrates the existing `OptimismPortal` tests over to the `OptimismPortal2` and adds new tests for the changed functionality.

**Metadata**
closes https://github.com/ethereum-optimism/client-pod/issues/506

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9399** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9399?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #9400** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9400?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #9401** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9401?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9401?utm_source=stack-comment).

File: packages/contracts-bedrock/invariant-docs/OptimismPortal2.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+**Test:** [`OptimismPortal2.t.sol#L182`](../test/invariants/OptimismPortal2.t.sol#L182)
+
+A withdrawal that has been proven should not be able to be finalized until after the finalization period has elapsed. 
+
+## `finalizeWithdrawalTransaction` should revert if the withdrawal has already been finalized.
+**Test:** [`OptimismPortal2.t.sol#L212`](../test/invariants/OptimismPortal2.t.sol#L212)
+
+Ensures that there is no chain of calls that can be made that allows a withdrawal to be finalized twice. 
+
+## A withdrawal should **always** be able to be finalized `FINALIZATION_PERIOD_SECONDS` after it was successfully proven.
```
Comment (smartcontracts) [inline packages/contracts-bedrock/invariant-docs/OptimismPortal2.md]: These invariants need to be updated

Comment (clabby) [reply packages/contracts-bedrock/invariant-docs/OptimismPortal2.md]: Updated docs 👍

**Review by smartcontracts — COMMENTED** [review]
Makes me realize the require for `OptimismPortal: withdrawal hash has already been proven, and dispute game is not invalid` should also check if the game type of the proven game has changed. Another way in which the game can become invalid.

File: packages/contracts-bedrock/test/Specs.t.sol [hunk 2]
```diff
@@ -235,6 +235,7 @@ contract Specification_Test is CommonTest {
         });
         _addSpec({ _name: "OptimismPortal2", _sel: _getSel("finalizedWithdrawals(bytes32)") });
         _addSpec({ _name: "OptimismPortal2", _sel: _getSel("guardian()") });
+        _addSpec({ _name: "OptimismPortal2", _sel: _getSel("sauron()") });
```
[the remark this answers is not in the document]
Comment (Inphi) [reply packages/contracts-bedrock/test/Specs.t.sol:239]: The rabbit is half-right. We need to introduce a `SAURON` role and use it for the `disputeGameBlacklist`, `deleteProvenWithdrawal` and `setRespectedGameType` specs

**Review by Inphi — APPROVED** [review]
LGTM!

**Pull request merged**